client/post-editor/restore-post-dialog.jsx

```diff
@@ -1,84 +1,89 @@
+		translate: PropTypes.func,
+		onClose: PropTypes.func,
+		onRestore: PropTypes.func,
+		isAutosave: PropTypes.bool,
+		postType: PropTypes.string,
+		userCanDeletePost: PropTypes.bool,
+	}
 
-	propTypes: {
-		post: React.PropTypes.object.isRequired,
-		onClose: React.PropTypes.func,
-		onRestore: React.PropTypes.func,
-		isAutosave: React.PropTypes.bool
-	},
+	static defaultProps = {
+		onClose: noop,
+		onRestore: noop,
+		isAutosave: false,
+	}
 
-	restorePost() {
-		if ( this.props.isAutosave ) {
-			this.props.onRestore();
-		} else if ( utils.userCan( 'delete_post', this.props.post ) ) {
-			this.props.onRestore( 'draft' );
+	restorePost = () => {
+		const { isAutosave, userCanDeletePost, onRestore } = this.props;
+		if ( isAutosave ) {
+			onRestore();
+		} else if ( userCanDeletePost ) {
+			onRestore( 'draft' );
 		}
-	},
+	}
 
-	getDialogButtons() {
+	getDialogButtons = () => {
+		const { translate, onClose } = this.props;
 		return [
 			<FormButton
 				key="restore"
 				isPrimary={ true }
 				onClick={ this.restorePost }>
-					{ this.translate( 'Restore' ) }
+					{ translate( 'Restore' ) }
 			</FormButton>,
 			<FormButton
 				key="back"
 				isPrimary={ false }
-				onClick={ this.props.onClose }>
-					{ this.translate( "Don't restore" ) }
+				onClick={ onClose }>
+					{ translate( 'Don\'t restore' ) }
 			</FormButton>
 		];
-	},
+	}
 
-	getStrings() {
-		if ( this.props.isAutosave ) {
-			if ( utils.isPage( this.props.post ) ) {
+	getStrings = () => {
+		const { isAutosave, postType, translate } = this.props;
+		const isPage = postType === 'page';
+		if ( isAutosave ) {
+			if ( isPage ) {
 				return {
-					dialogTitle: this.translate( 'Saved Draft' ),
-					dialogContent: this.translate( 'A more recent revision of this page exists. Restore?' ),
+					dialogTitle: translate( 'Saved Draft' ),
+					dialogContent: translate( 'A more recent revision of this page exists. Restore?' ),
 				};
 			}
 			return {
-				dialogTitle: this.translate( 'Saved Draft' ),
-				dialogContent: this.translate( 'A more recent revision of this post exists. Restore?' ),
+				dialogTitle: translate( 'Saved Draft' ),
+				dialogContent: translate( 'A more recent revision of this post exists. Restore?' ),
 			};
 		}
-		if ( utils.isPage( this.props.post ) ) {
+		if ( isPage ) {
 			return {
-				dialogTitle: this.translate( 'Deleted Page' ),
-				dialogContent: this.translate( 'This page has been sent to the trash. Restore it to continue writing.' ),
+				dialogTitle: translate( 'Deleted Page' ),
+				dialogContent: translate( 'This page has been sent to the trash. Restore it to continue writing.' ),
 			};
 		}
 		return {
-			dialogTitle: this.translate( 'Deleted Post' ),
-			dialogContent: this.translate( 'This post has been sent to the trash. Restore it to continue writing.' ),
+			dialogTitle: translate( 'Deleted Post' ),
+			dialogContent: translate( 'This post has been sent to the trash. Restore it to continue writing.' ),
 		};
```

Comment:
I've never seen a `switch` used in this way and I think its awesome but I also think it kills readability a bit